pkg/agent/controller/egress/ipassigner/ip_assigner_linux.go

```diff
@@ -129,14 +127,16 @@ func (a *ipAssigner) AssignIP(ip string) error {
 		return err
 	}
 
-	isIPv4 := parsedIP.To4()
-	if isIPv4 != nil {
-		if err := arping.GratuitousARPOverIface(isIPv4, a.externalInterface); err != nil {
+	if addr.IP.To4() != nil {
+		if err := arping.GratuitousARPOverIface(addr.IP.To4(), a.externalInterface); err != nil {
 			return fmt.Errorf("failed to send gratuitous ARP: %v", err)
 		}
 		klog.V(2).InfoS("Sent gratuitous ARP", "ip", parsedIP)
-	} else {
-		klog.ErrorS(ipv6NotSupportErr, "Failed to send Advertisement", "ip", parsedIP)
+	} else if addr.IP.To16() != nil {
```

Comment:
as we have verified the IP, maybe just use "else" to avoid a repeated conversion.

Reply:
Done

> I assume you have tested this PR in a IPv6 cluster and it worked? Have you checked whether the generated NDP packet can fresh an external Node's ip neighbor cache?

I have tested and I will put some test data here soon.

Reply:
The external Node can receive the NDP NA or GratuitousARP packets, but IP neighbor cache didn't fresh.

Reply:
IIRC, receiving GARP will only flush the stale entry and won't add a new entry, is this your observation? or the stale entry was not deleted? You mean both NDP NA and GARP didn't work?

Reply:
I think both NDP NA and GARP can work, sent message packets successfully. The integration test seems back to normal. @tnqn 

In both IPv4 and IPv6 cluster it will add a new entry finally: 
```
ip neigh show
2620:124:6020:1006:250:56ff:fea8:aaaa dev ens192  FAILED
```
ipv4
```
ip neigh show
10.176.26.123 dev ens192 lladdr 00:50:56:a7:36:c1 STALE
```

'10.176.26.123' is an IPv4 address assigned to an Egress Node.

"2620:124:6020:1006:250:56ff:fea8:aaaa" is an IPv6 address assigned to an Egress Node.

In IPv6 cluster, 
k get eg
```
NAME              EGRESSIP                                AGE   NODE
egress-web-ipv6   2620:124:6020:1006:250:56ff:fea8:aaaa   12m   wenqiq01-0
```

The IP has been assigned to Node 'wenqiq01-0'.

We can seen other Node received neighbor advertisement messages:

tcpdump -i ens192  icmp6
```
06:51:25.012447 IP6 fe80::250:56ff:fea7:36c1 > ip6-allnodes: ICMP6, neighbor advertisement, tgt is 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
06:51:25.013613 IP6 fe80::250:56ff:fea7:36c1 > ip6-allnodes: ICMP6, neighbor advertisement, tgt is 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
06:51:25.031403 IP6 fe80::250:56ff:fea7:36c1 > ip6-allnodes: ICMP6, neighbor advertisement, tgt is 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
```

ndp -i ens192 -a global listen
```
ndp listen> neighbor advertisement from fe80::250:56ff:fea7:36c1:
  - router:         false
  - solicited:      false
  - override:       true
  - target address: 2620:124:6020:1006:250:56ff:fea8:aaaa
  - options:
    - target link-layer address: 00:50:56:a7:36:c1
```

Ping from Pod(which has been selected by Egress podSelector) to external Node: 

tcpdump -i ens192  icmp6
```
06:53:05.157995 IP6 2620:124:6020:1006:250:56ff:fea8:aaaa > wenqiq01-2: ICMP6, echo request, seq 3, length 64
06:53:06.171124 IP6 wenqiq01-2 > ff02::1:ffa8:aaaa: ICMP6, neighbor solicitation, who has 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
06:53:06.181933 IP6 2620:124:6020:1006:250:56ff:fea8:aaaa > wenqiq01-2: ICMP6, echo request, seq 4, length 64
06:53:07.206164 IP6 2620:124:6020:1006:250:56ff:fea8:aaaa > wenqiq01-2: ICMP6, echo request, seq 5, length 64
06:53:07.206251 IP6 wenqiq01-2 > ff02::1:ffa8:aaaa: ICMP6, neighbor solicitation, who has 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
06:53:08.219153 IP6 wenqiq01-2 > ff02::1:ffa8:aaaa: ICMP6, neighbor solicitation, who has 2620:124:6020:1006:250:56ff:fea8:aaaa, length 32
```